core/iwasm/interpreter/wasm_runtime.h

```diff
@@ -79,8 +79,11 @@ typedef union {
 struct WASMMemoryInstance {
     /* Module type */
     uint32 module_type;
+
+    bool is_shared_memory;
+
     /* Shared memory flag */
-    bh_atomic_32_t ref_count; /* 0: non-shared, > 0: reference count */
+    bh_atomic_16_t ref_count; /* 0: non-shared, > 0: reference count */
```

Comment:
@wenyongh 
doesn't this break aot abi?
what size and alignment are you assuming for a bool here?

the whole purpose of having ref_count and is_shared_memory combined was to avoid aot abi.

Reply:
Pur my understanding, the bool size is 1 byte and the total size of WASMMemoryInstance is same as before, and ref_count is at 2-byte aligned address, it doesn't violate the aot abi. And the AOT code doesn't access the two fields.

There are also bh_static_asserts in aot_runtime.c:
https://github.com/bytecodealliance/wasm-micro-runtime/blob/main/core/iwasm/aot/aot_runtime.c#L37-L46

Reply:
iirc, standard-wise, a bool can be larger.
for things where the size is important, it's simpler to use uint8 or 16.

Reply:
OK, let's create another PR to use uint8 instead.